server/dive_server/crud_rpc.py

```diff
@@ -377,6 +378,35 @@ def postprocess(
 
     if not skipJobs and not isClone:
         token = Token().createToken(user=user, days=2)
+
+        # extract ZIP Files if not already completed
+        zipItems = Folder().childItems(
+            dsFolder,
+            filters={
+                "lowerName": {"$regex": constants.zipRegex},
+            },
+        )
+        # only support one zip file found
+        for item in zipItems:
+            # if item is already extracted create/move into 'source' folder
+            if constants.ZipFileExtractedMarker in item['meta'].keys():
+                source_folder = crud.get_or_create_source_folder(dsFolder, user)
+                Item().move(item, source_folder)
+                continue
```

Comment:
It might make sense to do this as part of the girder worker task: once extract is complete, have girder worker make a move request to the server.  That way you don't need this new metadata at all.

Reply:
you weren't thinking of a creating a specific endpoint, right?
More of a create folder, copy item and delete original?:
- `POST /folder`
- `POST /item/id/copy` - (I don't think there is a move)
- `DELETE /item/id/`
or
- `GirderClient.createFolder('sourceFolder')`
- `Girderclient.uploadFile(zipFile)` -again no Move/Copy for Girder Client
- `GirderClient.delete(olditemId)`

Reply:
No new endpoint is needed.  PUT `/item/{id}` will move an item to a different location.

https://viame.kitware.com/api/v1#!/item/item_updateItem